flaps/flaps.go

```diff
@@ -590,6 +596,36 @@ func (f *Client) sendRequest(ctx context.Context, method, endpoint string, in, o
 	return nil
 }
 
+var flapsCallRegex = regexp.MustCompile(`\/(?P<machineId>[a-z-A-Z0-9]*)\/(?P<flapsCall>[a-zA-Z0-9]*)`)
+
+type flapsCall struct {
+	Call       string  `json:"c"`
+	Command    string  `json:"co"`
+	Duration   float64 `json:"d"`
+	StatusCode int     `json:"s"`
+}
+
+func sendFlapsCallMetric(ctx context.Context, endpoint string, timing instrument.CallTimer, statusCode int) {
+
+	matches := flapsCallRegex.FindStringSubmatch(endpoint)
+	index := flapsCallRegex.SubexpIndex("flapsCall")
+
+	// unless something changes about flaps, this should always be false (meaning the regex passes)
+	if index >= len(matches) {
+		err := fmt.Errorf("flaps endpoint %s did not match regex %s", endpoint, flapsCallRegex.String())
+		sentry.CaptureException(err)
+		return
+	}
+
+	call := matches[index]
+	metrics.Send(ctx, "flaps_call", flapsCall{
+		Call:       call,
+		Command:    command_context.FromContext(ctx).Name(),
```

Comment:
two questions:
what would a value for call look like, and what values does `command_context.FromContext(ctx).Name()` give?

for call, I'm worried about cardinality, and for Command, I'm worried that both `fly apps restart` and `fly machines restart` would both return `restart`.

Reply:
`call` should always be the basic flaps call that's being done, like `lease` or `wait`. Any extra data like `?instance_id` should get filtered out by the regex.

As for both `apps restart` and `machines restart` returning restart, you're right, that's a bug I missed, and I'll fix that now.